webpack/__mocks__/foremanReact/common/I18n.js

```diff
@@ -1 +1,2 @@
 export const translate = val => val;
+export const sprintf = val => val;
```

Comment:
Any reason this needs to be mocked? I imagine the real implementation should just work.

Reply:
I don't know why, the whole thing is confusing to me, but without it `npm test` fails with:

```
 FAIL  webpack/components/PreupgradeReports/__tests__/PreupgradeReports.test.js
  PreupgradeReports
    ✕ encountered a declaration exception (70 ms)

  ● PreupgradeReports › encountered a declaration exception

    TypeError: (0 , _I18n.sprintf) is not a function

      120 |         key="preupgrade-reports-error"
      121 |         icontype="error-circle-o"
    > 122 |         msg={sprintf(__('Could not retrieve data: %(status) - %(msg)'), {
          |              ^
      123 |           status: error.statusText,
      124 |           msg: error.errorMsg,
      125 |         })}

      at webpack/components/PreupgradeReports/PreupgradeReports.js:122:14
      at ReactShallowRenderer.render (node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:829:32)
```

And adding `import { sprintf, translate as __ } from 'foremanReact/common/I18n';` to tests doesn't have any effect at all.

Reply:
Oh, it probably mocks out the whole `_I18n` module. I don't know enough about the JS side to know if this is right.

@MariaAga any advice here?

Reply:
Created WIP PR for JS stuff: https://github.com/theforeman/foreman_leapp/pull/125

Reply:
yeah this mock is fine, we usually do something similar in plugins

Reply:
It surprises me because the usual signature of sprintf is `def sprintf(str, *args)` and then it returns a single str. Is `val => val` doing the same thing?

Reply:
it depends what the tests are testing, it used to be `val => val` in rex but when I needed to do specifically test for something that used sprintf I changed it to `export const sprintf = (string, ...args) => [string, ...args].join(' + ');`